### PR TITLE
Support classical Fisher gradients when using TF and torch

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Previously the classical Fisher information transform was added, but gave incorrect gradients for some interfaces. The Autograd interface gradient was fixed in https://github.com/PennyLaneAI/pennylane/pull/2688. This PR fixes gradients for the TensorFlow and PyTorch interfaces.

**Description of the Change:**
Add support for classical Fisher gradients for TF and torch. Also removed the `pytest.mark.xfail` mark for the corresponding tests.

**Benefits:**
Differentiating the classical Fisher information now works for all interfaces 😄 .

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
N/A

### CODE REVIEW

File: pennylane/qinfo/transforms.py [hunk 1]
```diff
@@ -181,8 +181,8 @@ def _torch_jac(circ):
     def wrapper(*args, **kwargs):
         loss = functools.partial(circ, **kwargs)
         if len(args) > 1:
-            return torch.autograd.functional.jacobian(loss, (args))
-        return torch.autograd.functional.jacobian(loss, *args)
+            return torch.autograd.functional.jacobian(loss, args, create_graph=True)
```
Comment (Qottmann) [inline pennylane/qinfo/transforms.py:184]: what does this do?

File: tests/qinfo/test_fisher.py [hunk 2]
```diff
@@ -412,9 +412,9 @@ def circ(x, y, z):
                 qml.RZ(zi, wires=1)
             return qml.probs(wires=range(n_wires))
 
-        x = [tf.Variable(np.pi / 8) for _ in range(2)]
-        y = [tf.Variable(np.pi / 8) for _ in range(10)]
-        z = [tf.Variable(1.0) for _ in range(1)]
+        x = tf.Variable(np.pi / 8 * np.ones(2), trainable=True)
```
Comment (Qottmann) [inline tests/qinfo/test_fisher.py:415]: Ah nice!

File: pennylane/qinfo/transforms.py [hunk 3]
```diff
@@ -385,14 +381,14 @@ def wrapper(*args, **kwargs):
         p = new_qnode(*args, **kwargs)
 
         # In case multiple variables are used, we create a list of cfi matrices
-        if isinstance(j, tuple) and len(j) > 1:
+        if isinstance(j, tuple):
             res = []
             for j_i in j:
-                if interface == "tf":
```
[the remark this answers is not in the document]
Comment (eddddddy) [reply pennylane/qinfo/transforms.py:391]: I think for those specific tests you were using a list of small tensors instead of one big tensor, e.g. `[tf.Variable(np.pi/8) for _ in range(10)]` instead of `tf.Variable(np.pi/8 * np.ones(10))`. With the latter, the shape is the same as all the other interfaces.

File: pennylane/qinfo/transforms.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/qinfo/transforms.py:184]: The issue was that the value of the Jacobian wasn't being "watched", which caused the gradient computation later to be incorrect. Adding `create_graph=True` creates a path back to the input from the Jacobian, which fixes it.

File: tests/qinfo/test_fisher.py (hunk 2 above)
Comment (eddddddy) [reply tests/qinfo/test_fisher.py:415]: Yep, for the other interface tests (and now TF as well), we use a big tensor instead of a list of small tensors.

File: pennylane/qinfo/transforms.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Qottmann) [reply pennylane/qinfo/transforms.py:391]: great! Thanks for checking that

File: pennylane/qinfo/transforms.py (hunk 1 above)
Comment (Qottmann) [reply pennylane/qinfo/transforms.py:184]: Ah okay so in torch I have to specify that I want the jacobian to be differentiable (watched)? Really strange that it computed something of the right shape then but with wrong values ...

Comment (eddddddy) [reply pennylane/qinfo/transforms.py:184]: Right, the gradient was still computed because the contribution from the QNode probabilities was non-zero, but there was no contribution from the Jacobian term.

File: pennylane/qinfo/transforms.py [hunk 4]
```diff
@@ -361,6 +357,30 @@ def circ(params):
...
+
+    >>> print(qml.qinfo.classical_fisher(circ)(params))
+    (tensor([[0.13340679, 0.03650311],
+             [0.03650311, 0.00998807]], requires_grad=True)
+    >>> print(qml.jacobian(qml.qinfo.classical_fisher(circ))(params))
+    array([[[9.98030491e-01, 3.46944695e-18],
+            [1.36541817e-01, 5.15248592e-01]],
+
+            [[1.36541817e-01, 5.15248592e-01],
+            [2.16840434e-18, 2.81967252e-01]]]))
```
Comment (josh146) [inline pennylane/qinfo/transforms.py:382]:
```suggestion
            [1.36541817e-01, 5.15248592e-01]],
           [[1.36541817e-01, 5.15248592e-01],
            [2.16840434e-18, 2.81967252e-01]]]))
```

Since the line break breaks the rendering

**Review by josh146 — APPROVED** [review]
Thanks @eddddddy, this is great!

File: tests/qinfo/test_fisher.py [hunk 5]
```diff
@@ -530,11 +528,13 @@ def circ(params):
     @pytest.mark.torch
     @pytest.mark.tf
     @pytest.mark.jax
-    @pytest.mark.xfail
     def test_consistency(self):
```
Comment (rmoyard) [inline tests/qinfo/test_fisher.py:531]: This test is skip on the CI checks, you need to mark this one with `@pytest.mark.all_interfaces`

Comment (rmoyard) [reply tests/qinfo/test_fisher.py:531]: @eddddddy You can change that and then merge 👍
